hbp_validation_framework/__init__.py

```diff
@@ -338,10 +344,12 @@ def _download_resource(self, uuid):
             raise Exception("The provided 'uuid' is invalid!")
         else:
             data = data.json()
+            dirpath = target_loc
             if data["entity_type"] == "folder":
-                if not os.path.exists(data["name"]):
-                    os.makedirs(data["name"])
-                os.chdir(data["name"])
+                dirpath = os.path.join(target_loc, data["name"])
+                if not os.path.exists(dirpath):
+                    os.makedirs(dirpath)
+                os.chdir(dirpath)
```

Comment:
I don't understand why you are changing directory here.

Reply:
Yes, this was avoidable. Have updated the target directory specification such that this is no longer required.